server_test.go

```diff
@@ -2,13 +2,204 @@ package veneur
 
 import (
 	"bytes"
+	"compress/zlib"
+	"encoding/json"
 	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
+// set up a boilerplate local config for later use
+func localConfig() Config {
+	return Config{
+		APIHostname:         "http://localhost",
+		Debug:               false,
+		Hostname:            "localhost",
+		Interval:            100 * time.Millisecond,
+		Key:                 "",
+		MetricMaxLength:     4096,
+		Percentiles:         []float64{.5, .75, .99},
+		ReadBufferSizeBytes: 2097152,
+		UDPAddr:             "localhost:8126",
+		HTTPAddr:            "localhost:8127",
+		ForwardAddr:         "http://localhost",
+		NumWorkers:          96,
+		NumReaders:          4,
+		StatsAddr:           "localhost:8125",
+		Tags:                []string{},
+		SentryDSN:           "",
+		FlushLimit:          1024,
+	}
+}
+
+// assertMetrics checks that all expected metrics are present
+// and have the correct value
+func assertMetrics(t *testing.T, metrics DDMetricsRequest, expectedMetrics map[string]float64) {
+	// it doesn't count as accidentally quadratic if it's intentional
+	for metricName, expectedValue := range expectedMetrics {
+		assertMetric(t, metrics, metricName, expectedValue)
+	}
+}
+
+func assertMetric(t *testing.T, metrics DDMetricsRequest, metricName string, value float64) {
+	defer func() {
+		if r := recover(); r != nil {
+			assert.Fail(t, "error extracting metrics", r)
+		}
+	}()
+	for _, metric := range metrics.Series {
+		if metric.Name == metricName {
+			assert.Equal(t, int(value+.5), int(metric.Value[0][1]+.5), metricName)
```

Comment:
nit: instead of rounding, we can use an epsilon assertion here (which compares by units of least place, if I recall correctly). One unfortunate trap of that comparison type is that it doesn't work on zero though.

more fun details on this subject: https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/